warehouse/packaging/metadata_verification.py

```diff
@@ -41,6 +50,86 @@ def _verify_url_pypi(url: str, project_name: str, project_normalized_name: str)
     )
 
 
+def _get_url_content(url: rfc3986.api.URIReference, max_length_bytes: int) -> str:
+    """
+    Get at least `max_length_bytes` of the contents of a URL
+
+    Raises `requests.exception.RequestException` on any errors while trying to
+    access the URL.
+    """
+    headers = {"User-Agent": "warehouse/1.0 (PyPI URL verifier, https://pypi.org)"}
+    r = requests.get(
+        url.unsplit(),
+        allow_redirects=False,
+        timeout=5,
+        stream=True,
+        headers=headers,
+    )
+    r.raise_for_status()
+
+    content = next(r.iter_content(max_length_bytes))
```

Comment:
I _think_ `r.raw.read(max_length_bytes)` will be slightly faster + more idiomatic here, since we're already setting `stream=True` 🙂

Reply:
Fixed!